### DISCUSSION TITLE
Need Help with OVRDBF Commands Using Code for IBM i in VS Code

### DISCUSSION THREAD

**Discussion by antoine59910** [codefori/vscode-ibmi#1776] [Q&A] [answered]
Hello everyone,

I am currently working on an IBM i project using the "Code for IBM i" extension in Visual Studio Code. I've encountered a specific problem for which I am seeking advice or solutions from the community.

Problem:
When trying to compile my RPGLE/SQLRPGLE programs, I need to execute OVRDBF (Override Database File) commands prior to the compilation to redirect certain files. However, even after executing these commands in VS Code, it seems that the compilation fails, indicating that the overrides have not been executed correctly or are not being recognized.

Questions:

Is there a specific method or recommended steps for executing OVRDBF commands via "Code for IBM i" in VS Code?
How can I verify in VS Code that my OVRDBF commands have been successfully executed before proceeding with compilation?
Are there any particular configurations or settings in "Code for IBM i" or VS Code that I should check or adjust?
Any help, advice, or experiences would be greatly appreciated. I am particularly interested in workflows or configurations that have worked for others in similar scenarios.

I had modify already the "action.js" to add : 
{
    "extensions": ["GLOBAL"],
    "name": "Override Database File (OVRDBF)",
    "command": "OVRDBF FILE(${filename|Nom du fichier à rediriger}) TOFILE(${tofilename|Nom du nouveau fichier ou membre})",
    "deployFirst": true,
    "environment": "ile"
  }

It seems to do the action but when i compiled my SQLRPGLE program, it seems the OVRDBF doesn't worked...
I also tried to do this from a SQL file with : 
cl:OVRDBF FILE(RUTFENF1) TOFILE(RUTFEN03);
cl:OVRDBF FILE(RGBFLIF1) TOFILE(RGBFLI01);
cl:OVRDBF FILE(RGBFENF1) TOFILE(RGBFEN01);
cl:OVRDBF FILE(RFUFENF1) TOFILE(RFUFEN01);

Thank you in advance for your time and assistance!

Best regards,

**Comment by chrjorgensen** [comment, marked as answer]
Hi @antoine59910 

The Code for IBM i runs your commands stateless and in a new job. Thus your OVRDBF has no effect on the following commands.

It is documented on the Action maintenance screen:

![image](https://github.com/codefori/vscode-ibmi/assets/13275072/198c8a60-fa2c-42d5-ab20-1d289a62b687)

One solution could be to implement the [BUILD utility by Scott Klement](https://www.scottklement.com/build/). This can be used to store the commands used to create the object in the source of the object. I have used this for years with great success.

There are probably other methods besides this, though...
